portlist/portlist.go

```diff
@@ -18,6 +18,27 @@ type Port struct {
 	inode string // OS-specific; "socket:[165614651]" on Linux
 }
 
+// IsInteresting reports whether p is an interesting enough port
+// to report to our peer nodes for discovery purposes.
+func (p *Port) IsInteresting() bool {
+	if p.Proto == "udp" {
+		return false
+	}
+	switch p.Port {
+	case 22, // ssh
+		80,   // http
+		443,  // https
+		3389, // rdp
+		5900: // vnc
+		return true
+	}
+	if p.Port >= 8000 && p.Port <= 8999 {
+		// Likely some HTTP dev server.
+		return true
+	}
```

Comment:
I don't think this wildcard will work reliably. It still means 1/64th of randomly generated port numbers will generate service map updates, which on a machine with lots of churn (Chrome and/or Windows, yay) could still be a problem. I also wouldn't feel confident offering "just any port starting with 8" as an http:// link by default in a UI, since it might be a random Chrome listener, so the long list doesn't help us build the user-facing product AFAICS.

In the future we could have a CLI or GUI for exporting ports (bonus points: let the user actually describe the service being offered), and/or probe them to make sure they're valid http/https before publishing. In the meantime, I'd be more comfortable having a very short list of auto-published ports and documentation that says basically "if you want your http server to show up, run it on 80, 443 or 8080."

(I guess 443 won't even work right since we don't know what hostname to use)

If we just want the full port list available for debugging, I suggest just printing a log line with all the numbers whenever portlist gives a new list. That won't cause any tailcontrol churn.

It's easier to add more ports in the future than to take them away.

Reply:
Yeah, I also wasn't super keen on adding policy to the portlist package. Policy in ipn SGTM. Will move.

I agree we should probe new ports when they come into existence and see what they are before reporting them to tailcontrol.

I'm still with @crawshaw on not liking the whitelist in general. We _can_ document that 8080 is magic, but it kills the magic of it just working for whatever port you're using. I suppose we can do that later, though.

Sure, will restrict to 8080 for now. At least it'll motivate me to do a better thing sooner.

Reply:
Ok, good.

I'm not kidding about the log lines; it won't be too spammy to do (iOS, the battery sensitive one, has basically no visible listeners to our sandbox, and certainly not ones that change a lot) and then you'll have all the information you ever needed to tune the magic however you want.

Reply:
Do random ports really get assigned in 8000-9000?

According to https://en.wikipedia.org/wiki/Ephemeral_port, there's well-defined behavior on various OSs. The Linux kernel assigns ephemeral ports from a declared range:

```
$ cat /proc/sys/net/ipv4/ip_local_port_range
32768	60999
```

It looks like Windows Server 2008 might use ports in the [8000,8999] range but nothing else does.

Logs would be very useful here to let us tune this later on, especially if the logs give us a sense of how long-lived ports are. (I suspect there's an interesting category of short-lived ports that keep coming back, i.e. my compile-run loop bash command has the same port number each time.)